app/src/main/java/org/jellyfin/androidtv/ui/SettingsPopup.kt

```diff
@@ -0,0 +1,73 @@
+package org.jellyfin.androidtv.ui
+
+import android.content.Context
+import android.graphics.Color
+import android.graphics.drawable.ColorDrawable
+import android.view.Gravity
+import android.view.LayoutInflater
+import android.view.View
+import android.widget.PopupWindow
+import org.jellyfin.androidtv.databinding.SettingsPopupBinding
+import org.jellyfin.androidtv.util.Utils
+
+class SettingsPopup(
+	context: Context,
+	anchor: View,
+	audioChangedListener: ValueChangedListener<Long>?,
+	subtitleChangedListener: ValueChangedListener<Long>?
+) {
+	val popupWindow: PopupWindow?
+	private val anchor: View
+	private val binding: SettingsPopupBinding
+
+	init {
+		val width = Utils.convertDpToPixel(context, width)
+		val height = Utils.convertDpToPixel(context, fullHeight)
+
+		binding = SettingsPopupBinding.inflate(LayoutInflater.from(context), null, false)
+
+		popupWindow = PopupWindow(binding.root, width, height)
```

Comment:
assign directly on variable declaration instead. Use .apply{} to set the values from the next few lines

Reply:
TIL. moved the entire init function to the constructor.